src/main/java/com/celements/mandatory/XWikiRecycleBinIndexes.java

```diff
@@ -0,0 +1,41 @@
+package com.celements.mandatory;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.Requirement;
+
+import com.celements.query.IQueryExecutionServiceRole;
+import com.xpn.xwiki.XWikiException;
+
+@Component(XWikiRecycleBinIndexes.NAME)
+public class XWikiRecycleBinIndexes implements IMandatoryDocumentRole {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(XWikiRecycleBinIndexes.class);
+
+  public static final String NAME = "celements.mandatory.addIndexToRecycleBin";
+
+  @Requirement
+  IQueryExecutionServiceRole queryExecService;
+
+  @Override
+  public List<String> dependsOnMandatoryDocuments() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public void checkDocuments() throws XWikiException {
+    LOGGER.info("executing XWikiRecycleBinIndexes");
+    if (queryExecService.existsConstraint("prefix_db", "xwikirecyclebin", "dateIDX")) {
```

Comment:
- a `!` (NOT) is missing
- I only used `prefix_db` as placeholder, fill it with the actual database and prefix

Reply:
are you sure about the ! (NOT)?